projects/plugins/jetpack/extensions/shared/components/product-management-controls/inspector-control.js

```diff
@@ -53,7 +56,15 @@ export default function ProductManagementInspectorControl( {
 		event.preventDefault();
 		setApiState( API_STATE_REQUESTING );
 		saveProduct(
-			{ title, currency, price, interval },
+			{
+				title,
+				currency,
+				price,
+				interval,
+				type: isMarkedAsDonation ? 'donation' : null,
+				buyer_can_change_amount: isCustomAmount,
+				public: true,
```

Comment:
If going through with the suggestion in https://github.com/Automattic/jetpack/pull/23714#issuecomment-1085622255 to rename public, this will need updating too

Reply:
Done :)